File: android/app/src/main/java/io/v/syncslides/model/Deck.java
// Copyright 2015 The Vanadium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package io.v.syncslides.model;

import android.graphics.Bitmap;

/**
 * A deck, aka a set of slides.
 */
public interface Deck {
    /**
     * Returns a Bitmap suitable as a thumbnail of the deck (e.g. the title
     * slide).
     */
    Bitmap getThumb();

    /**
     * Returns raw thumbnail data.
     */
    byte[] getThumbData();

    /**
     * Returns the title of the deck.
     */
    String getTitle();

    /**
     * Returns the deck id.
     */
    String getId();
}

File: dart/lib/models/deck.dart
// Copyright 2015 The Vanadium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

import 'dart:convert';

import 'blobref.dart';

// Deck represents a deck of slides.
class Deck {
  Deck(this._key, this._name, this._thumbnail);

  Deck.fromJson(String key, String json) {
    Map map = JSON.decode(json);
    _key = key;
    _name = map['name'];
    _thumbnail = new BlobRef(map['thumbnailkey']);
  }

  String _key;
  String get key => _key;

  String _name;
  String get name => _name;

  BlobRef _thumbnail;
  BlobRef get thumbnail => _thumbnail;

  String toJson() {
    // NOTE(aghassemi): We never serialize the key with the object.
    Map map = new Map();
    map['name'] = name;
    map['thumbnailkey'] = thumbnail.key;
    return JSON.encode(map);
  }

  // TODO(aghassemi): Override == and hash
}
